plugins/postgres/dbt/adapters/postgres/relation.py

```diff
@@ -20,3 +34,57 @@ def __post_init__(self):
 
     def relation_max_name_length(self):
         return 63
+
+    def get_materialized_view_config_change_collection(
+        self, relation_results: RelationResults, runtime_config: RuntimeConfigObject
+    ) -> Optional[PostgresMaterializedViewConfigChangeCollection]:
+        """
+        Postgres-specific implementation of `BaseRelation.get_materialized_view_config_changes`.
+
+        The only tracked changes for materialized views are indexes.
+        """
+        config_change_collection = PostgresMaterializedViewConfigChangeCollection()
+
+        existing_materialized_view = PostgresMaterializedViewConfig.from_relation_results(
+            relation_results
+        )
+        new_materialized_view = PostgresMaterializedViewConfig.from_model_node(
+            runtime_config.model
+        )
+
+        if index_config_changes := self._get_index_config_changes(
+            existing_materialized_view.indexes, new_materialized_view.indexes
+        ):
+            config_change_collection.indexes = index_config_changes
+
+        if config_change_collection.has_changes:
+            return config_change_collection
```

Comment:
if there is _any_ change, we return the entire collection, right? Is this what downstream callers expect? Or do they assume that all config changes in a non-null PostgresMaterializedViewConfigChangeCollection from this method are changes

Reply:
Yes, we return the entire collection either way, because we might need to inspect the collection to know what to do. For example, if there are changes that don't require a full refresh, and changes that do, I don't want to apply changes and then refresh anyway. And all changes should be changes, if I understand that correctly.